modules/admin_manual/pages/configuration/server/security/hsmdaemon.adoc

```diff
@@ -0,0 +1,588 @@
+        Token present:    yes
+        Flags:
+    Token info:
+        Manufacturer ID:    SoftHSM project
+        Model:              SoftHSM v2
+        Hardware version:   2.2
+        Firmware version:   2.2
+        Serial number:      e8ba06bca0c9daa6
+        Initialized:        yes
+        User PIN init.:     yes
+        Label:              oc token without pin
+        MaxSessionCount:    0
+        SessionCount:       18446744073709551615
+        MaxRwSessionCount:  0
+        RwSessionCount:     18446744073709551615
+        MaxPinLen:          255
+        MinPinLen:          4
+        TotalPublicMemory:  18446744073709551615
+        FreePublicMemory:   18446744073709551615
+        TotalPrivateMemory: 18446744073709551615
+        FreePrivateMemory:  18446744073709551615
+        UTCTime:            2019021408270200
+        Flags: CKF_RNG CKF_LOGIN_REQUIRED CKF_RESTORE_KEY_NOT_NEEDED CKF_USER_PIN_COUNT_LOW
+Slot: 1989683358,
+    Slot info:
+        Description:      SoftHSM slot ID 0x7698289e
+        Manufacturer ID:  SoftHSM project
+        Hardware version: 2.2
+        Firmware version: 2.2
+----
+
+TIP: See the {opensc-wiki-url}[OpenSC Wiki] for more information.
+
+==== Configure the Slot and Pin
+
+Ask the customer which slot to use and if a PIN is needed. 
+Update `/etc/hsmdaemon/hsmdaemon.toml` with the information that the customer provides, in the `pkcs11` section, as in the example below.
+
+....
+[pkcs11]
+module = "/usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so" # softhsm v2
+pin = "1234"          # The user pin supplied when running softhsm2-util --init-token, comment it out , or leave empty if no pin is necessary
+slot = 1989683358     # Find your slot id with `sudo hsmdaemon listslots`
+....
+
+==== Test Key Generation
+
+[NOTE] 
+====
+If no PIN is supplied, generating a new key might be protected by an operator card that has to be inserted in the HSM. 
+In this case, coordinate testing and final master key generation with your HSM team.
+====
+
+For testing key generation, run the command `hsmdaemon genkey test`, as in the following example.
+
+[source,console]
+----
+$ hsmdaemon genkey test
+Id: 9bac3719-2b8d-11e9-aeab-0242b5ece4c3, label: test
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAl1BO4vsI+xDk+x0nccl7
+HQhMR/hwfa0+N8fyYNI8yzTTmYDqz9aaF20qG48+mjC0AUEt2kfKo94xM3UeEw4c
+st4j1dpRJtmAJThcuN8OH3sa+3MeXWgGuWxjB1lxEEOqax2A6XzllDlbDsogwkOL
+hSkUU9AaMRBtF8fASJGtJDP+iXwdb7OsFg78PS1wBAISYSUwk06xY7LwWIxge+hY
+4oU+5x4itusdO6rz6kbcJtmUyDUb8DhKnN6OdkhnifUZLBG9HQyTa5OM+BAabbFZ
+mTM2gZlUnGKXN7c4kaBPFt1IfjjVYu7pvj3B2uxUf4GywuSuWGWnAy89FqeXteRV
+jwIDAQAB
+-----END PUBLIC KEY-----
+----
+
+==== Test Data Encryption
+
+For testing data encryption, run the `hsmdaemon encrypt` command, as in the following example.
+
+[source,console]
+----
+The first argument is the "Id:" value from running the genkey command above.
+$ sudo hsmdaemon encrypt 9bac3719-2b8d-11e9-aeab-0242b5ece4c3 Zm9vYmFy
```

Comment:
what is the second value `Zm9vYmFy` ?
where does it come from?

Reply:
Well spotted. It's the base64-encoded data to be encrypted.